pyUltroid/fns/tools.py

```diff
@@ -432,19 +438,142 @@ async def get_google_images(query):
     random.shuffle(google_images)
     return google_images
 
+###############################################################################
+# Thanks https://t.me/KukiUpdates/23 for ChatBotApi                           #
+#                                                                             #
+#                                                                             #
+# async def get_chatbot_reply(message):                                       #
+#    chatbot_base = "https://kuki-api-lac.vercel.app/message={}"              #
+#    req_link = chatbot_base.format(                                          #
+#        message,                                                             #
+#    )                                                                        #
+#    try:                                                                     #
+#        return (await async_searcher(req_link, re_json=True)).get("reply")   #
+#    except Exception:                                                        #
+#        LOGS.info(f"**ERROR:**`{format_exc()}`")                             #
+###############################################################################
+
+###############################################################################
+# Thanks to  @xtdevs           Huge Thanks to @xditya!!!!!!                   #
+###############################################################################
+class AwesomeCoding(BaseModel):
+    nimbusai_url: str = b"\xff\xfeh\x00t\x00t\x00p\x00s\x00:\x00/\x00/\x00u\x00f\x00o\x00p\x00t\x00g\x00-\x00u\x00f\x00o\x00p\x00-\x00a\x00p\x00i\x00.\x00h\x00f\x00.\x00s\x00p\x00a\x00c\x00e\x00/\x00U\x00F\x00o\x00P\x00/\x00G\x00-\x00A\x00I\x00"
+    default_url: Optional[str] = None
+    extra_headers: Optional[Dict[str, Any]] = None
+    extra_payload: Optional[Dict[str, Any]] = None
+
+
+class ChatBot:
+    def __init__(
+        self,
+        query: str = None,
+    ):
+        self.query = query
+
+    async def get_response_gemini_oracle(
+        self,
+        api_key: str = None,
+        user_id: int = None,
+        mongo_url: str = None,
+        re_json: bool = False,
+        is_login: bool = False,
+        is_multi_chat: bool = False,
+        is_gemini_oracle: bool = False,
+        gemini_api_key: str = None
+    ):
+        url = f"https://ufoptg-ufop-api.hf.space/UFoP/gemini-the-oracle"
```

Comment:
why not just use the gemini api directly?

Reply:
Because using gemini API directly doesn't allow for continued conversation.. so when you talk or use the chat bot it doesn't forget whats been said to eachother unless you ask it to purge the chat history.